packages/cart/src/services/cart-module.ts

```diff
@@ -681,4 +689,200 @@ export default class CartModuleService implements ICartModuleService {
+    @MedusaContext() sharedContext: Context = {}
+  ) {
+    const adjustments = await this.shippingMethodAdjustmentService_.list(
+      filters,
+      config,
+      sharedContext
+    )
+
+    return await this.baseRepository_.serialize<
+      CartTypes.ShippingMethodAdjustmentDTO[]
+    >(adjustments, {
+      populate: true,
+    })
+  }
+
+  @InjectTransactionManager("baseRepository_")
+  async setShippingMethodAdjustments(
+    cartId: string,
+    adjustments: (
+      | CartTypes.CreateShippingMethodAdjustmentDTO
+      | CartTypes.UpdateShippingMethodAdjustmentDTO
+    )[],
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<CartTypes.ShippingMethodAdjustmentDTO[]> {
+    const cart = await this.retrieve(
+      cartId,
+      { select: ["id"], relations: ["shipping_methods.adjustments"] },
+      sharedContext
+    )
+
+    const methodIds = cart.shipping_methods?.map((method) => method.id)
+
+    const existingAdjustments = await this.listShippingMethodAdjustments(
+      { shipping_method_id: methodIds },
+      { select: ["id"] },
+      sharedContext
+    )
+
+    let toUpdate: CartTypes.UpdateShippingMethodAdjustmentDTO[] = []
+    let toCreate: CartTypes.CreateShippingMethodAdjustmentDTO[] = []
+    for (const adj of adjustments) {
+      if ("id" in adj) {
+        toUpdate.push(adj as CartTypes.UpdateShippingMethodAdjustmentDTO)
+      } else {
+        toCreate.push(adj as CartTypes.CreateShippingMethodAdjustmentDTO)
+      }
+    }
+
+    const adjustmentsSet = new Set(toUpdate.map((a) => a.id))
+
+    const toDelete: CartTypes.ShippingMethodAdjustmentDTO[] = []
+
+    // From the existing adjustments, find the ones that are not passed in adjustments
+    existingAdjustments.forEach(
+      (adj: CartTypes.ShippingMethodAdjustmentDTO) => {
+        if (!adjustmentsSet.has(adj.id)) {
+          toDelete.push(adj)
+        }
+      }
+    )
+
+    await this.shippingMethodAdjustmentService_.delete(
+      toDelete.map((adj) => adj!.id),
+      sharedContext
+    )
+
+    let result: ShippingMethodAdjustment[] = []
+
+    if (toCreate.length) {
+      const created = await this.shippingMethodAdjustmentService_.create(
+        toCreate,
+        sharedContext
+      )
+
+      result.push(...created)
+    }
+
+    if (toUpdate.length) {
+      const updated = await this.shippingMethodAdjustmentService_.update(
+        toUpdate,
+        sharedContext
+      )
+      result.push(...updated)
+    }
+
+    return await this.baseRepository_.serialize<
+      CartTypes.ShippingMethodAdjustmentDTO[]
+    >(result, {
+      populate: true,
+    })
+  }
+
+  async addShippingMethodAdjustments(
+    adjustments: CartTypes.CreateShippingMethodAdjustmentDTO[]
+  ): Promise<CartTypes.ShippingMethodAdjustmentDTO[]>
+  async addShippingMethodAdjustments(
+    adjustment: CartTypes.CreateShippingMethodAdjustmentDTO
+  ): Promise<CartTypes.ShippingMethodAdjustmentDTO[]>
```

Comment:
```suggestion
  ): Promise<CartTypes.ShippingMethodAdjustmentDTO>
```

Discard away if this is intentional, but it does look lil weird that it returns an array when the input is not